src/Workleap.Extensions.OpenApi.Analyzers/CompareTypedResultWithAnnotationAnalyzer.cs

```diff
@@ -0,0 +1,279 @@
+            var returnType = methodSymbol.ReturnType;
+
+            if (returnType is INamedTypeSymbol typedReturnType)
+            {
+                var iResultTypeSymbol = context.Compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Http.IResult");
+                if (iResultTypeSymbol is null)
+                {
+                    return;
+                }
+
+                // Check if the return type is of Task<IResult> or Task<Result<>>. If yes, then keep the inner type.
+                if (SymbolEqualityComparer.Default.Equals(typedReturnType.ConstructedFrom, this.TaskOfTSymbol))
+                {
+                    var subType = typedReturnType.TypeArguments[0];
+                    if (subType is INamedTypeSymbol namedSubType)
+                    {
+                        typedReturnType = namedSubType;
+                    }
+                }
+
+                if (Implements(typedReturnType, iResultTypeSymbol))
+                {
+                    var methodSignatureStatusCodeToTypeMap = this.GetMethodReturnStatusCodeToTypeMap(typedReturnType);
+
+                    foreach (var attribute in methodSymbol.GetAttributes())
+                    {
+                        this.ValidateAnnotationWithTypedResult(context, attribute, methodSignatureStatusCodeToTypeMap);
+                    }
+                }
+            }
+        }
+
+        private void ValidateAnnotationWithTypedResult(SymbolAnalysisContext context, AttributeData attribute,
+            Dictionary<int, ITypeSymbol> methodSignatureStatusCodeToTypeMap)
+        {
+            if (attribute.AttributeClass != null)
+            {
+                if (attribute.AttributeClass.Equals(this.ProducesResponseSymbol, SymbolEqualityComparer.Default))
+                {
+                    if (attribute.ConstructorArguments.Length == 1)
+                    {
+                        if (attribute.ConstructorArguments[0].Value is int statusCodeValue)
+                        {
+                            var type = this._statusCodeToResultsMap[statusCodeValue];
+                            ValidateAnnotationForTypeMismatch(context, methodSignatureStatusCodeToTypeMap, statusCodeValue, type, attribute);
+                        }
+                    }
+                    else
+                    {
+                        if (attribute.ConstructorArguments[1].Value is int statusCodeValue && attribute.ConstructorArguments[0].Value is ITypeSymbol type)
+                        {
+                            ValidateAnnotationForTypeMismatch(context, methodSignatureStatusCodeToTypeMap, statusCodeValue, type, attribute);
+                        }
+                    }
+                }
+
+                else if (attribute.AttributeClass.ConstructedFrom.Equals(this.ProducesResponseOfTSymbol, SymbolEqualityComparer.Default))
+                {
+                    if (attribute.ConstructorArguments[0].Value is int statusCodeValue)
```

Comment:
assert that the collection contains at least 1 item. (idem for other cases)

BTW, I think the conditions could be simplify if you use collection pattern matching. However, I think it requires some polyfill. You could add them manually, or use packages such as Polyfill or Meziantou.Polyfill.

````
if (attribute.ConstructorArguments is [{Value: int statusCode}])
````

Reply:
I think for now, will check if length == 0 at the top & return if equal, will also look into polyfill